src/main/webapp/app/layouts/main/main.module.ts

```diff
@@ -0,0 +1,13 @@
+import { CUSTOM_ELEMENTS_SCHEMA, NgModule } from '@angular/core';
+
+import { ArTEMiSSharedModule } from 'app/shared';
+import { JhiMainComponent } from 'app/layouts';
+
+@NgModule({
+    imports: [ArTEMiSSharedModule],
+    declarations: [JhiMainComponent],
+    entryComponents: [],
+    exports: [JhiMainComponent],
+    schemas: [CUSTOM_ELEMENTS_SCHEMA],
+})
+export class ArTEMiSMainModule {}
```

Comment:
The `tutor-exercise-dashboard.spec.ts` needs the `JhiMainComponent` for whatever reason. Probably because it's listed in the `entryComponents` of the tutor exercise and course dashboard. I also tried to remove the `entryComponents` entry there, but it did not make any difference. I could not figure out another way of importing the `JhiMainModule` so far. Maybe you have any idea @thilo-behnke?

Reply:
In my opinion the module declaration of the TutorExerciseDashboard was incorrect, listing entry components of other modules.
I made a commit to restructure the module so that it uses its own components as entry components. For me angular could still compile the application and I could run your tests.